### ISSUE TITLE
select alignment issue

### ISSUE THREAD

**Issue by svantreeck** [cefsharp/CefSharp#906]
Hi, I've been having an issue where select tags seem to have their dropdowns misaligned using bootstrap 3.  The code below seems to illustrate this in the Wpf Example project.  Any suggestions on fixing this would be appreciated.  Thanks!

```
<!DOCTYPE html>
<html>
<head lang="en">
    <meta charset="UTF-8">
    <title>Sample</title>
    <!-- Latest compiled and minified CSS -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap.min.css">

    <!-- Optional theme -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap-theme.min.css">


</head>
<body>
    <div class="container">
        <div class="row">
            <div class="col-lg-4 form-group">
                <label class="control-label">
                    Test Select
                </label>
                <select class="form-control">
                    <option value="0" label="0">0</option>
                    <option value="1" label="1">1</option>
                    <option value="2" label="2">2</option>
                    <option value="3" label="3">3</option>
                    <option value="4" label="4">4</option>
                </select>
            </div>
        </div>
    </div>

    <script src="http://code.jquery.com/jquery-2.1.3.min.js" />
    <!-- Latest compiled and minified JavaScript -->
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
</body>
</html>
```

**Comment by amaitland** [comment]
Can you please confirm the versions you've tested with. Also html code is best posted as a fiddler link.

**Comment by amaitland** [comment]
(Should have been `jsfiddle` link).

**Comment by amaitland** [comment]
- Can you provide a `jsfiddle` link that reproduces the problem
- Does the problem exist in the `WinForms` example?
- Are you using a `DPI > 1`?
- What versions have you tested with?
- Is it just bootstrap dropdowns that are the problem?

**Comment by svantreeck** [comment]
Here's a JSFiddle:
https://jsfiddle.net/4be895dw/

I cloned the Wpf.Example project to reproduce it there (using cef 3.2171.1979) on the 24th, using DPI=1.  I'll rerun using the winform example tomorrow when I have the computer again.  

When I tried using the Wpf Example on a different computer with 150% DPI scaling, the dropdowns were positioned properly (although with some other artifacts).  I'm wondering if my problem might be Graphics Driver related?  I haven't seen similar before, but I'll retest with winforms and see what I can find there.  Anything else you'd look at?

**Comment by amaitland** [comment]
> Anything else you'd look at?

Start with that. As it stands I can't reproduce the problem. What sort of graphics card are you using? What operating system?

**Comment by svantreeck** [comment]
I was able to test the winforms example and it looks fine.  I was also able to reproduce using WPF with a normal select. 

I was digging a bit more and the computer is a laptop running intel hd 4000 graphics with a driver from windows update circa 2014...I'm going to try updating it and see if it resolves the problems I'm seeing.

**Comment by svantreeck** [comment]
Updating graphics drivers didn't seem to affect this.  

I've found an interesting pattern that seems like the offset of the dropdown is based on the width of the previous dropdown that was opened.  For example, say dropdown A is 100px wide and dropdown B is 200px wide.  If I open dropdown A twice in successtion, the right edge of the popup will be aligned with the left edge of the dropdown.  Now if I go and open dropdown B, the first time its opened (after dropdown A), it will be offset by 100px.  If I close it and reopen it, it gets offset by 200px (the width of itself).  Now if I go back to dropdown A, when I open it the first time, its popup is offset by 200px, and the second time i open it, it goes back to 100px.

**Comment by amaitland** [comment]
`Intel` graphics and `CEF` just don't seem to play nicely, very common to have issues with them. Unfortunately `CEF` doesn't expose the same graphics detection that comes with `Chrome`.

If you open `chrome://gpu` in `Chrome`, you will usually see a `Problems Detected` section, might give come more info. (Every graphics seems to have one problem or another, `Intel` just has more).

It's usually worth trying to disable gpu acceleration.
https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Example/CefExample.cs#L32

It may be nothing to do with graphics, it's just a place to start.

**Comment by svantreeck** [comment]
I'm going to close it for now...
I was doing some debugging through the `SetPopupSizeAndPositionImpl` method and the inputs were the same despite the visual positioning changing.  

I was able to reproduce the popup positioning issue outside of CefSharp, so I'm going to do some digging in the Wpf forums about this to see if there's something else going on.

**Comment by amaitland** [comment]
Windows does strange things at times, like http://stackoverflow.com/questions/18113597/wpf-handedness-with-popups so it's possible something else is at play.

**Comment by svantreeck** [comment]
Yeah that's exactly the issue...was able to fix it with that, but it seems a bit hackish by using reflection to set it.  I'm going to see if I can fix it by trying some different positioning settings on the popup and will create a pull request if I can find one that works in all situations.

**Comment by svantreeck** [comment]
fixed with pull request #916

**Comment by svantreeck** [comment]
It looks fine to me on win 8.1, I think its good to go unless you can see any issues with it.
